src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -161,8 +142,10 @@ private void addListingToBusinesses(PlacesSearchResult place, String url) {
     double rating = place.rating;
     Photo photos[] = place.photos;
     String types[] = place.types;
-    allBusinesses.add(new Listing(name, formattedAddress, 
-          placeLocation, rating, photos, types, url));
+    Listing listing = new Listing(name, formattedAddress, 
+          placeLocation, rating, photos, types);
+    listing.addPlaceId(place.placeId);
```

Comment:
Why not do this in the constructor, following suit of the other attributes?

Reply:
I moved it to the constructor - I was avoiding changing all of the files again to add this as a parameter, but it's now fixed!

Reply:
That is a good point. Sounds like this is a design cleanup that can be implemented. This would avoid adding unnecessary code in objects where it's irrelevant. I don't believe this is a particular issue at the moment, but as the application grows this refactor may be worth thinking about.